packages/http-client-csharp/emitter/src/lib/client-model-builder.ts

```diff
@@ -62,33 +62,53 @@ export function createModel(sdkContext: SdkContext<NetEmitterOptions>): CodeMode
     Clients: inputClients,
     Auth: processServiceAuthentication(sdkContext, sdkPackage),
   };
+
   return clientModel;
 
   function fromSdkClients(
+    sdkContext: SdkContext<NetEmitterOptions>,
     clients: SdkClientType<SdkHttpOperation>[],
     inputClients: InputClient[],
     parentClientNames: string[],
   ) {
     for (const client of clients) {
-      const inputClient = emitClient(client, parentClientNames);
+      const inputClient = fromSdkClient(sdkContext, client, parentClientNames);
       inputClients.push(inputClient);
       const subClients = client.methods
         .filter((m) => m.kind === "clientaccessor")
         .map((m) => m.response as SdkClientType<SdkHttpOperation>);
       parentClientNames.push(inputClient.Name);
-      fromSdkClients(subClients, inputClients, parentClientNames);
+      fromSdkClients(sdkContext, subClients, inputClients, parentClientNames);
       parentClientNames.pop();
     }
   }
 
-  function emitClient(client: SdkClientType<SdkHttpOperation>, parentNames: string[]): InputClient {
+  function fromSdkClient(
+    sdkContext: SdkContext<NetEmitterOptions>,
+    client: SdkClientType<SdkHttpOperation>,
+    parentNames: string[],
+  ): InputClient {
     const endpointParameter = client.initialization.properties.find(
       (p) => p.kind === "endpoint",
     ) as SdkEndpointParameter;
     const uri = getMethodUri(endpointParameter);
     const clientParameters = fromSdkEndpointParameter(endpointParameter);
+    const clientName = getClientName(client, parentNames);
+    // see if this namespace is a sub-namespace of an existing bad namespace
+    const segments = client.clientNamespace.split(".");
+    const lastSegment = segments[segments.length - 1];
+    if (lastSegment === clientName) {
+      // this segment is bad
+      reportDiagnostic(sdkContext.program, {
+        code: "client-namespace-conflict",
+        format: { clientNamespace: client.clientNamespace, clientName },
+        target: client.__raw.type ?? NoTarget,
+      });
+    }
+
     return {
-      Name: getClientName(client, parentNames),
+      Name: clientName,
+      ClientNamespace: client.clientNamespace,
```

Comment:
Should this just be `Namespace` given that it is on the `InputClient` type?

Reply:
we name this `clientnamespace` because TCGC name this `clientNamespace` and we do not want to invent a new name.
It might be confusing but the `clientNamespace` means its namespace in the client, like `client name` (the decorator) not the namespace of a client

Reply:
> It might be confusing but the clientNamespace means its namespace in the client, like client name (the decorator) not the namespace of a client

Not sure I see the distinction. Do you mean it meant to differentiate from the tsp concept of namespace?

Reply:
here I just would like to follow the same name whatever TCGC has, at least in our JSON. Because we have a final goal that in the future we do not create our own interfaces but directly serialize whatever we got from TCGC.
I opened an issue to suggest them to rename it: https://github.com/Azure/typespec-azure/issues/2123

I think in our C# input models, we could change it to `Namespace`.

Reply:
all ClientNamespace properties in C# types have been changed to `Namespace`.